packages/nimble-components/src/table/components/cell/tests/table-cell.spec.ts

```diff
@@ -0,0 +1,125 @@
+import { html, css } from '@microsoft/fast-element';
+import { TableCell } from '..';
+import { waitForUpdatesAsync } from '../../../../testing/async-helpers';
+import { fixture, type Fixture } from '../../../../utilities/tests/fixture';
+import type { TableCellRecord, TableCellState } from '../../../types';
+import { TableCellPageObject } from './table-cell.pageobject';
+
+interface SimpleTableCellRecord extends TableCellRecord {
+    stringData: string;
+}
+
+// prettier-ignore
+async function setup(): Promise<Fixture<TableCell<SimpleTableCellRecord>>> {
+    return fixture<TableCell<SimpleTableCellRecord>>(
+        html`<nimble-table-cell>
+            </nimble-table-cell>`
+    );
+}
+
+describe('TableCell', () => {
+    let element: TableCell<SimpleTableCellRecord>;
+    let connect: () => Promise<void>;
+    let disconnect: () => Promise<void>;
+    let pageObject: TableCellPageObject<SimpleTableCellRecord>;
+
+    beforeEach(async () => {
+        ({ element, connect, disconnect } = await setup());
+        pageObject = new TableCellPageObject<SimpleTableCellRecord>(element);
+    });
+
+    afterEach(async () => {
+        await disconnect();
+    });
+
+    it('can construct an element instance', () => {
+        // prettier-ignore
+        expect(document.createElement('nimble-table-cell')).toBeInstanceOf(TableCell);
+    });
+
+    it('renders cellTemplate content', async () => {
+        await connect();
+
+        const cellTemplate = html` <span>Foo</span> `;
+
+        element.cellTemplate = cellTemplate;
+        await waitForUpdatesAsync();
+
+        const renderedContent = pageObject.getRenderedCellContent();
+        expect(renderedContent?.textContent).toBe('Foo');
+    });
+
+    it('changing cellState updates rendered content', async () => {
+        await connect();
+
+        const cellTemplate = html<TableCellState<SimpleTableCellRecord>>`
+            <span>${x => x.cellRecord.stringData}</span>
+        `;
+        element.cellState = {
+            cellRecord: { stringData: 'foo' }
+        } as TableCellState<SimpleTableCellRecord>;
```

Comment:
Sorry to tosss in a wrench, but there are a whole bunch of as casts we should try and avoid

Reply:
For example here you can probably do:
```ts
let someVar: TableCellState<SimpleTableCellRecord>  = {
            cellRecord: { stringData: 'foo' }
        };
element.cellState = someVar;
```
The difference is that TypeScript will typecheck the value being passed in the above example so if the type changes we will catch issues. With the as cast we aren't getting any benefit from typing.

If we need Mocks with partial implementations and want to use `as` with them (since we aren't actually implementing a fully valid type) we should make them very explicitly as mocks.

Reply:
Eh, I just supplied and empty `columnConfig` field.